src-go/server/intercept.go

```diff
@@ -0,0 +1,155 @@
+package server
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"sync"
+
+	"github.com/open-ch/ja3"
+)
+
+type interceptProxy struct {
+	burpAddr        string
+	m               sync.RWMutex
+	clientHelloData map[string]string
+	l               net.Listener
+}
+
+func newInterceptProxy(interceptAddr, burpAddr string) (*interceptProxy, error) {
+	l, err := net.Listen("tcp", interceptAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	s := interceptProxy{
+		burpAddr:        burpAddr,
+		m:               sync.RWMutex{},
+		clientHelloData: map[string]string{},
+		l:               l,
+	}
+
+	go s.start()
+
+	return &s, nil
+}
+
+func (s *interceptProxy) getTLSFingerprint(sni string) string {
+	s.m.RLock()
+	defer s.m.RUnlock()
+
+	return s.clientHelloData[sni]
+}
+
+func (s *interceptProxy) start() {
+	for {
+		conn, err := s.l.Accept()
+		if err != nil {
+			log.Println(err)
+		}
+
+		go s.handleConn(conn)
+	}
+}
+
+func (s *interceptProxy) handleConn(in net.Conn) {
+	defer in.Close()
+
+	out, err := net.Dial("tcp", s.burpAddr)
+	if err != nil {
+		return
+	}
+
+	defer out.Close()
+
+	var readClientHello bool
+	var length uint16
+	var clientHello []byte
+
+	r := io.TeeReader(in, out)
+
+	var wg sync.WaitGroup
+
+	wg.Add(2)
+
+	go func() {
+		for {
+			var buf []byte
+
+			buf = make([]byte, 1)
+			if _, err := r.Read(buf); err != nil {
+				return
```

Comment:
So, when there's an error it just fails silently? My question applies to the return statements below too.